platform/pulpcore/app/models/content.py

```diff
@@ -63,11 +92,33 @@ def storage_path(self, name):
         """
         return default_storage.get_artifact_path(self.sha256)
 
-    file = models.FileField(db_index=True, upload_to=storage_path, max_length=255)
-    size = models.IntegerField(blank=False, null=False)
-    md5 = models.CharField(max_length=32, blank=False, null=False, unique=False, db_index=True)
-    sha1 = models.CharField(max_length=40, blank=False, null=False, unique=False, db_index=True)
-    sha224 = models.CharField(max_length=56, blank=False, null=False, unique=False, db_index=True)
-    sha256 = models.CharField(max_length=64, blank=False, null=False, unique=True, db_index=True)
-    sha384 = models.CharField(max_length=96, blank=False, null=False, unique=True, db_index=True)
-    sha512 = models.CharField(max_length=128, blank=False, null=False, unique=True, db_index=True)
+    artifact = models.ForeignKey(Artifact, on_delete=models.CASCADE, null=True)
+    content = models.ForeignKey(Content, on_delete=models.CASCADE)
+    relative_path = models.CharField(max_length=64)
+
+
+class DeferredArtifact(Model):
```

Comment:
Suggest another name.  Perhaps `PendingArtifact` might be better since they are created even when downloading is not deferred.